src/d3a/models/area/__init__.py

```diff
@@ -134,10 +134,14 @@ def restore_state(self, saved_state):
         self.stats.restore_state(saved_state)
 
     def area_reconfigure_event(self, **kwargs):
-        if self.strategy is not None:
+        if key_in_dict_and_not_none(kwargs, 'strategy_object'):
+            self.strategy = kwargs['strategy_object']
+            self.strategy.area = self.parent
+            self.strategy.owner = self
```

Comment:
Aren't these already done by the `activate` method https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/area/__init__.py#L232 ?

Reply:
Yes, tested it and you are right.